splunk-devops/src/main/java/com/splunk/splunkjenkins/listeners/LoggingQueueListener.java

```diff
@@ -30,43 +34,57 @@
 @SuppressWarnings("unused")
 @Extension
 public class LoggingQueueListener extends QueueListener {
+    // Cache to keep track of queue time
     private final static Cache<Long, Float> cache = CacheBuilder.newBuilder()
             .maximumSize(3000).build();
+    //To keep track of blocked events timestamp
+    private final static ConcurrentHashMap<Long, Long> blockedQueue = new ConcurrentHashMap<Long, Long>();
```

Comment:
* Sorry for not adding comment for using a soft reference key in Map with maximumSize, per Jenkins doc, the queue can be cleared directly 
https://support.cloudbees.com/hc/en-us/articles/360051376772-How-can-I-purge-clean-the-build-queue-
```
Jenkins.instance.queue.clear()
```
clear method will not trigger onLeaveX listener (while `q.cancel(it.task)` does), so some items left in the cache for such cases. if we use `queueItems = new ConcurrentHashMap<Long, Long>()` to hold the items, it will grow bigger

* I think blocked/buildable/waiting are mutual exclusive, one item can not be at two different state at the same time, could we just use one map to hold blocked/buildable/waiting items?

Reply:
Thank you for your comments.

- Comment #1: I have verified the case where the queue could be cleared out using `Jenkins.instance.queue.clear()`. It looks like the OnLeaveX was called for all the different phases when I cleared the queue from Script Console. I was having 9 events during a particular duration and found 9 matching OnLeaveX events for the phases like waiting and blocked.

- Comment #2: Good catch. It makes sense to use one map for mutually exclusive items. I will work on the suggested changes.

Reply:
thanks